include/dali/c_api.h

```diff
@@ -96,24 +96,18 @@ DLL_PUBLIC void daliInitialize();
 
 /// @{
 /**
- * @brief Create DALI pipeline. Setting batch_size,
+ * @brief Create DALI pipeline. Setting max_batch_size,
```

Comment:
I don't understand the naming of this parameter. It's called max_batch_size, but there is no constrain on the External Input batch size provided by the user. Should be this parameter (and the corresponding field inside the Pipeline class) called something like *default batch size*? I don't see this *max* having any meaning here.

Reply:
My intention was that if user tries to feed batch size bigger than `max_batch_size`, he'll get an error. The reason is that DALI preallocates data according to this value, so setting it too high explodes memory consumption.

This strategy also exists in Triton, where you provide `max_batch_size` for given model. I guess it's for the same reason - to preallocate some of buffers needed

Reply:
Is there any `ENFORCE` for `batch_size < max_batch_size`? I can't see it anywhere.

Reply:
Good point. Added to `RunHelper`